query/services/ingestion.py

```diff
@@ -0,0 +1,199 @@
+                                if "\0" in tag_value:
+                                    logger.warning(
+                                        f"Product: {product_code}. Nuls stripped from {tag} value: {tag_value}"
+                                    )
+                                    tag_data[i] = tag_value.replace("\0", "")
+                        await connection.execute(
+                            "INSERT INTO product_temp (id, last_updated, data) VALUES ($1, $2, $3) ON CONFLICT DO NOTHING",
+                            existing_product["id"],
+                            last_updated,
+                            product_data,
+                        )
+
+                        update_count += 1
+                        if not (update_count % batch_size):
+                            await apply_staged_changes(
+                                connection, obsolete, update_count, process_id, source
+                            )
+
+                if update_count % batch_size:
+                    await apply_staged_changes(
+                        connection, obsolete, update_count, process_id, source
+                    )
+                if skip_count % batch_size:
+                    logger.info(
+                        f"Skipped {skip_count}{' obsolete' if obsolete else ''} products"
+                    )
+
+            # Mark all products specifically requested but not found as deleted
+            if codes_specified:
+                requested_product_codes = filter["code"]["$in"]
+                missing_product_codes = [
+                    code
+                    for code in requested_product_codes
+                    if code not in found_product_codes
+                ]
+                if missing_product_codes:
+                    await delete_products(
+                        connection, process_id, source, missing_product_codes
+                    )
+
+            if source == Source.full_load:
+                await delete_products(connection, process_id, source)
+                await append_loaded_tags(connection, TAG_TABLES.keys())
+        finally:
+            await connection.execute("DROP TABLE product_temp")
+
+    return max_last_updated
+
+
+async def apply_staged_changes(
+    connection: Connection, obsolete, update_count, process_id, source
+):
+    await connection.execute("ANALYZE product_temp")
+    log = logger.debug
+    await update_products_from_staging(connection, log, obsolete, process_id, source)
+    await create_ingredients_from_staging(connection, log, obsolete)
+    await create_tags_from_staging(connection, log, obsolete)
+    await fixup_product_countries(connection, obsolete)
+    await connection.execute("TRUNCATE TABLE product_temp")
+
+    logger.info(f"Imported {update_count}{' obsolete' if obsolete else ''} products")
+
+
+import_running = False
+
+
+async def import_from_mongo(from_date: str = None):
```

Comment:
```suggestion
async def import_from_mongo(from_date: str = None):
    """Import all products that where changed in mongodb since `from_data`
    
    This is noticeably useful for first products import,
    or in substitution to redis events, for regular imports.
```